pkg/daemon/ceph/osd/volume.go

```diff
@@ -262,13 +262,14 @@ func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *Device
 				devices.Entries["metadata"].Config.Name,
 			}...)
 
-			crushDeviceClass := os.Getenv(oposd.CrushDeviceClassVarName)
-			if crushDeviceClass != "" {
-				metadataArg = append(metadataArg, []string{crushDeviceClassFlag, crushDeviceClass}...)
-			}
 			metadataBlockPath = devices.Entries["metadata"].Config.Name
 		}
 
+		crushDeviceClass := os.Getenv(oposd.CrushDeviceClassVarName)
```

Comment:
This block needs to move below L273 in the condition `if device.Data == -1 {`.
Just append `[]string{crushDeviceClassFlag, crushDeviceClass}` to `immediateExecuteArgs`.